server/src/server/competition_info.py

```diff
@@ -0,0 +1,89 @@
+import copy
+import hashlib
+import os
+import typing
+import yaml
+
+from server_types import ChallengeInfoKeys
+
+
+class CompetitionInfo(object):
+    def __init__(self, data_dir: str, event: str):
+        self._event = event
+        event_dir = os.path.join(data_dir, event)
+        self._teams = load_teams(event_dir)
+        self._challenges = load_challenges(event_dir)
+        self._challenge_infos = load_challenge_infos(event_dir, self._challenges)
+
+    @property
+    def event(self) -> str:
+        return self._event
+
+    def list_teams(self) -> typing.List[str]:
+        return list(self._teams)
+
+    def list_challenges(self) -> typing.List[str]:
+        return list(self._challenges)
+
+    def get_challenge_info(self, challenge: str) -> dict:
+        if challenge in self._challenge_infos:
+            return copy.deepcopy(self._challenge_infos[challenge])
+        else:
+            return {
+                ChallengeInfoKeys.NAME: "",
+                ChallengeInfoKeys.DESCRIPTION: "",
+                ChallengeInfoKeys.SCORE_TABLE: [],
+                ChallengeInfoKeys.AVAILABLE_ATTEMPTS: [],
+            }
```

Comment:
Will this dict with keys, but empty values be filled later on? Otherwise I don't see the need to add the keys.

Reply:
If we're sending challenge info, I'd like it to stick to a 'default' format. The alternative I could think of was leaving it out altogether if no useful data. Then you could do something like I did in https://github.com/tue-robotics/at-home-refbox/commit/d91c9cb3f101f18152dfb68e96898c2171c94be0 . If you (also) prefer that we can cherry-pick it to this branch

Reply:
Although this is like a 'null object pattern'